### PR TITLE
Fix resolveService deprecation

### PR DESCRIPTION
**Pull request by RobozinhoD**
## Purpose / Description
Fix resolveService deprecation

## Fixes
Fixes part of #12364 

## Approach
Bases myself on #12422 and added a method to compat

## How Has This Been Tested?

I didn't know how to test it

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.kt [hunk 1]
```diff
@@ -83,5 +85,9 @@ class CompatHelper private constructor() {
         inline fun <reified T : Parcelable> Intent.getParcelableExtraCompat(name: String): T? {
             return compat.getParcelableExtra(this, name, T::class.java)
         }
+
+        fun PackageManager.resolveServiceCompat(intent: Intent, flags: Int): ResolveInfo? {
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.kt:89]: The name should be `resolveService` to follow the platform naming for easier migration. See #12550 for a similar case(for `resolveActivity`) where I explained how to tackle this.

**Review by lukstbit — CHANGES_REQUESTED** [review]
The newer version of `resolveService` uses the api level 33 class [ResolveInfoFlags](https://developer.android.com/reference/android/content/pm/PackageManager.ResolveInfoFlags). I think we should duplicate this functionality to easier migrate later to the platform code. This was almost done in #12550 (which also needs it) and I'm inclined to put this PR on hold until that PR is merged(so we can use the class introduced there).

Of course if you want to work on it now I can review it.

**Comment by RobozinhoD** [comment]
> I think we should duplicate this functionality to easier migrate later to the platform code

I find this harder

File: AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.kt (hunk 1 above)
Comment (RobozinhoD) [reply AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.kt:89]: warning "Extension is shadowed by a member: public abstract fun resolveService(intent: Intent, flags: Int): ResolveInfo?"

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.kt:89]: Resolving - valid reasoning

**Comment by david-allison** [comment]
Agreed with the above. See: https://github.com/ankidroid/Anki-Android/pull/13250 for an example

**Review by david-allison — CHANGES_REQUESTED** [review]

**Comment by RobozinhoD** [comment]
the example helped a lot, thx

**Review by david-allison — APPROVED** [review]
Awesome

**Review by mikehardy — APPROVED** [review]
David established the pattern prior and likes it - looks good to me as well, thank you!

**Pull request merged**